packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3199,15 +3210,16 @@ class StatefulElement extends ComponentElement {
   void update(StatefulWidget newWidget) {
     super.update(newWidget);
     assert(widget == newWidget);
-    final StatefulWidget oldConfig = _state._config;
-    // Notice that we mark ourselves as dirty before calling didUpdateConfig to
-    // let authors call setState from within didUpdateConfig without triggering
+    final StatefulWidget oldWidget = _state._widget;
+    // Notice that we mark ourselves as dirty before calling didUpdateWidget to
+    // let authors call setState from within didUpdateWidget without triggering
     // asserts.
     _dirty = true;
-    _state._config = widget;
+    _state._widget = widget;
     try {
       _debugSetAllowIgnoredCallsToMarkNeedsBuild(true);
-      _state.didUpdateConfig(oldConfig);
+      _state.didUpdateWidget(oldWidget);
+      _state.didUpdateConfig(oldWidget);
```

Comment:
Should I just remove the @Deprecated config then? If we're going to non backward compatible change, we might as well do it once.